project/Dependencies.scala

```diff
@@ -32,6 +32,7 @@ object Dependencies {
     "com.typesafe" % "config" % "1.4.2",
     "org.apache.logging.log4j" %% "log4j-api-scala" % "12.0",
     "org.apache.logging.log4j" % "log4j-api" % "2.17.2",
+    "org.apache.logging.log4j" % "log4j-core" % "2.17.2",
     "org.apache.logging.log4j" % "log4j-core" % "2.17.2" % "it,test",
```

Comment:
I believe line 36 is extraneous now that you've added line 35.  If log4j-core is available in compile scope, it'll also be available in test and integration test scopes too.

Reply:
log4j-core was intentionally not part of core. The assumption was that users that get Daffodil via maven can add whatever log4j implementation as a dependency that they want. This way we don't force a particular implementation.

With the tdml-runner now using code that depends on log4j-core, that may not be possible now. I'm not sure if there's away around this, but maybe it's not a big deal if log4j-core is now a hard dependency for Daffodil?

Note that if we remove line 36, we can also remove line 50 in the cli dependency (our CLI does bundle log4j-core).

Reply:
Do we need log4j-core for daffodil-core test, or only daffodil-tdml-lib or even maybe only daffodil-tdml-processor?

Reply:
I think we only need it for the new `TDMLAppender`, which is defined and used in `daffodil-tdml-lib`. So I think only lib needs this new log4j-core dependency.

Though you raise a good point, I think daffodil-tdml-procesor should be the thing aggregating logs, since different DFDL implementations will have different mechanisms for capturing logs. Daffodil uses log4j, but IBM DFDL might use something completely different. Seems maybe the new TDMLAppender wants to be moved to daffodil-tdml-processor, and the `DaffodilTDMLDFDLProcessor` must return the logs as part of the Parse/Unparse result and the TDMLRunner can then compare against the expected results?

Reply:
Note that adding log4j-core as a dependency to the daffodil-tdml-proessor has a good side effect: Our schema projects that depend on daffodil-tdml-processor for running tests will now transitively pull in log4j-core, and it should get rid of that warning about not having a logger available.

Reply:
I don't think this comment was addressed. log4-core should only be a dependency to the daffodil-tdml-processor project, not as a core dependency. And the TDMLAppender and related code also wants to be moved to the same project. Other DFDL implemenations (e.g. IBM DFDL) may use a different mechanism for their logs.